diff --git a/mujoco_ros2_control/tests/CMakeLists.txt b/mujoco_ros2_control/tests/CMakeLists.txt
index ef9f0b7..028655d 100644
--- a/mujoco_ros2_control/tests/CMakeLists.txt
+++ b/mujoco_ros2_control/tests/CMakeLists.txt
@@ -1,22 +1,30 @@
 find_package(ament_cmake_gtest REQUIRED)
 find_package(ament_cmake_pytest REQUIRED)
+find_package(rcl_logging_spdlog REQUIRED)
+get_filename_component(RCL_LOGGING_SPDLOG_PREFIX
+  "${rcl_logging_spdlog_DIR}/../../.." ABSOLUTE)
+set(RCL_LOGGING_SPDLOG_LIBRARY_DIR "${RCL_LOGGING_SPDLOG_PREFIX}/lib")
 
-ament_add_gtest(test_plugin test_plugin.cpp)
+ament_add_gtest(test_plugin test_plugin.cpp
+  APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}")
 target_link_libraries(test_plugin
   mujoco_ros2_control
   rclcpp::rclcpp
 )
 
-ament_add_gtest(test_headless_init test_headless_init.cpp)
+ament_add_gtest(test_headless_init test_headless_init.cpp
+  APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}")
 target_link_libraries(test_headless_init
   mujoco_ros2_control
   rclcpp::rclcpp
   hardware_interface::hardware_interface
 )
 
-ament_add_gtest(test_viewer_camera test_viewer_camera.cpp)
+ament_add_gtest(test_viewer_camera test_viewer_camera.cpp
+  APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}")
 target_link_libraries(test_viewer_camera
   mujoco_ros2_control
 )
 
-ament_add_pytest_test(test_urdf_to_mujoco_utils test_urdf_to_mujoco_utils.py)
+ament_add_pytest_test(test_urdf_to_mujoco_utils test_urdf_to_mujoco_utils.py
+  APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}")
diff --git a/mujoco_ros2_control_plugins/CMakeLists.txt b/mujoco_ros2_control_plugins/CMakeLists.txt
index 5800ce3..3866f01 100644
--- a/mujoco_ros2_control_plugins/CMakeLists.txt
+++ b/mujoco_ros2_control_plugins/CMakeLists.txt
@@ -62,9 +62,14 @@ install(TARGETS mujoco_ros2_control_plugins
 
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
+  find_package(rcl_logging_spdlog REQUIRED)
+  get_filename_component(RCL_LOGGING_SPDLOG_PREFIX
+    "${rcl_logging_spdlog_DIR}/../../.." ABSOLUTE)
+  set(RCL_LOGGING_SPDLOG_LIBRARY_DIR "${RCL_LOGGING_SPDLOG_PREFIX}/lib")
 
   ament_add_gtest(test_external_wrench_plugin
     test/test_external_wrench_plugin.cpp
+    APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}"
   )
   target_link_libraries(test_external_wrench_plugin
     mujoco_ros2_control_plugins
