diff --git a/mujoco_ros2_control/tests/CMakeLists.txt b/mujoco_ros2_control/tests/CMakeLists.txt
index 028655d..d425652 100644
--- a/mujoco_ros2_control/tests/CMakeLists.txt
+++ b/mujoco_ros2_control/tests/CMakeLists.txt
@@ -1,19 +1,33 @@
 find_package(ament_cmake_gtest REQUIRED)
 find_package(ament_cmake_pytest REQUIRED)
 find_package(rcl_logging_spdlog REQUIRED)
+find_package(rmw_fastrtps_cpp REQUIRED)
+find_package(rmw_dds_common REQUIRED)
 get_filename_component(RCL_LOGGING_SPDLOG_PREFIX
   "${rcl_logging_spdlog_DIR}/../../.." ABSOLUTE)
 set(RCL_LOGGING_SPDLOG_LIBRARY_DIR "${RCL_LOGGING_SPDLOG_PREFIX}/lib")
+get_filename_component(RMW_FASTRTPS_CPP_PREFIX
+  "${rmw_fastrtps_cpp_DIR}/../../.." ABSOLUTE)
+set(RMW_FASTRTPS_CPP_LIBRARY_DIR "${RMW_FASTRTPS_CPP_PREFIX}/lib")
+get_filename_component(RMW_DDS_COMMON_PREFIX
+  "${rmw_dds_common_DIR}/../../.." ABSOLUTE)
+set(RMW_DDS_COMMON_LIBRARY_DIR "${RMW_DDS_COMMON_PREFIX}/lib")
 
 ament_add_gtest(test_plugin test_plugin.cpp
-  APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}")
+  APPEND_LIBRARY_DIRS
+    "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}"
+    "${RMW_FASTRTPS_CPP_LIBRARY_DIR}"
+    "${RMW_DDS_COMMON_LIBRARY_DIR}")
 target_link_libraries(test_plugin
   mujoco_ros2_control
   rclcpp::rclcpp
 )
 
 ament_add_gtest(test_headless_init test_headless_init.cpp
-  APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}")
+  APPEND_LIBRARY_DIRS
+    "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}"
+    "${RMW_FASTRTPS_CPP_LIBRARY_DIR}"
+    "${RMW_DDS_COMMON_LIBRARY_DIR}")
 target_link_libraries(test_headless_init
   mujoco_ros2_control
   rclcpp::rclcpp
@@ -21,10 +35,16 @@ target_link_libraries(test_headless_init
 )
 
 ament_add_gtest(test_viewer_camera test_viewer_camera.cpp
-  APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}")
+  APPEND_LIBRARY_DIRS
+    "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}"
+    "${RMW_FASTRTPS_CPP_LIBRARY_DIR}"
+    "${RMW_DDS_COMMON_LIBRARY_DIR}")
 target_link_libraries(test_viewer_camera
   mujoco_ros2_control
 )
 
 ament_add_pytest_test(test_urdf_to_mujoco_utils test_urdf_to_mujoco_utils.py
-  APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}")
+  APPEND_LIBRARY_DIRS
+    "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}"
+    "${RMW_FASTRTPS_CPP_LIBRARY_DIR}"
+    "${RMW_DDS_COMMON_LIBRARY_DIR}")
diff --git a/mujoco_ros2_control_plugins/CMakeLists.txt b/mujoco_ros2_control_plugins/CMakeLists.txt
index 31317fb..175a54a 100644
--- a/mujoco_ros2_control_plugins/CMakeLists.txt
+++ b/mujoco_ros2_control_plugins/CMakeLists.txt
@@ -63,13 +63,24 @@ install(TARGETS mujoco_ros2_control_plugins
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
   find_package(rcl_logging_spdlog REQUIRED)
+  find_package(rmw_fastrtps_cpp REQUIRED)
+  find_package(rmw_dds_common REQUIRED)
   get_filename_component(RCL_LOGGING_SPDLOG_PREFIX
     "${rcl_logging_spdlog_DIR}/../../.." ABSOLUTE)
   set(RCL_LOGGING_SPDLOG_LIBRARY_DIR "${RCL_LOGGING_SPDLOG_PREFIX}/lib")
+  get_filename_component(RMW_FASTRTPS_CPP_PREFIX
+    "${rmw_fastrtps_cpp_DIR}/../../.." ABSOLUTE)
+  set(RMW_FASTRTPS_CPP_LIBRARY_DIR "${RMW_FASTRTPS_CPP_PREFIX}/lib")
+  get_filename_component(RMW_DDS_COMMON_PREFIX
+    "${rmw_dds_common_DIR}/../../.." ABSOLUTE)
+  set(RMW_DDS_COMMON_LIBRARY_DIR "${RMW_DDS_COMMON_PREFIX}/lib")
 
   ament_add_gtest(test_external_wrench_plugin
     test/test_external_wrench_plugin.cpp
-    APPEND_LIBRARY_DIRS "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}"
+    APPEND_LIBRARY_DIRS
+      "${RCL_LOGGING_SPDLOG_LIBRARY_DIR}"
+      "${RMW_FASTRTPS_CPP_LIBRARY_DIR}"
+      "${RMW_DDS_COMMON_LIBRARY_DIR}"
   )
   target_link_libraries(test_external_wrench_plugin
     mujoco_ros2_control_plugins
