From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "zjumty@aliyun.com" <zjumty@aliyun.com>
Date: Fri, 14 Aug 2026 16:33:38 +0800
Subject: [PATCH 08/11] platform C++17 requirements


diff --git a/mujoco_ros2_control/CMakeLists.txt b/mujoco_ros2_control/CMakeLists.txt
index 00a13b5..57a5bc1 100644
--- a/mujoco_ros2_control/CMakeLists.txt
+++ b/mujoco_ros2_control/CMakeLists.txt
@@ -2,6 +2,10 @@ cmake_minimum_required(VERSION 3.16)
 
 project(mujoco_ros2_control)
 
+if(APPLE)
+  enable_language(OBJCXX)
+endif()
+
 find_package(ros2_control_cmake REQUIRED)
 
 # find dependencies
@@ -116,11 +120,18 @@ else()
   set_target_properties(platform_ui_adapter PROPERTIES
   POSITION_INDEPENDENT_CODE ON
   )
+  target_compile_features(platform_ui_adapter PUBLIC cxx_std_17)
   target_sources(
     platform_ui_adapter
     PRIVATE ${MUJOCO_SIMULATE_DIR}/glfw_adapter.h ${MUJOCO_SIMULATE_DIR}/glfw_dispatch.h ${MUJOCO_SIMULATE_DIR}/platform_ui_adapter.h
     PRIVATE ${MUJOCO_SIMULATE_DIR}/glfw_adapter.cc ${MUJOCO_SIMULATE_DIR}/glfw_dispatch.cc ${MUJOCO_SIMULATE_DIR}/platform_ui_adapter.cc
   )
+  if(APPLE)
+    target_sources(platform_ui_adapter
+      PUBLIC ${MUJOCO_SIMULATE_DIR}/glfw_corevideo.h
+      PRIVATE ${MUJOCO_SIMULATE_DIR}/glfw_corevideo.mm)
+    target_link_libraries(platform_ui_adapter PUBLIC "-framework CoreVideo")
+  endif()
   target_include_directories(platform_ui_adapter PUBLIC ${MUJOCO_SIMULATE_DIR} ${MUJOCO_INCLUDE_DIR})
   target_link_libraries(platform_ui_adapter PRIVATE ${MUJOCO_LIB} glfw)
   target_compile_options(platform_ui_adapter PRIVATE ${MUJOCO_SILENCE_COMPILER_WARNINGS})
@@ -131,10 +142,15 @@ else()
     OUTPUT_NAME simulate
     POSITION_INDEPENDENT_CODE ON
   )
+  target_compile_features(mujoco_simulate PUBLIC cxx_std_17)
   target_sources(mujoco_simulate
     PRIVATE ${MUJOCO_SIMULATE_DIR}/simulate.h
     PRIVATE ${MUJOCO_SIMULATE_DIR}/simulate.cc ${MUJOCO_SIMULATE_DIR}/array_safety.h
   )
+  if(APPLE)
+    target_sources(mujoco_simulate PRIVATE ${MUJOCO_SIMULATE_DIR}/macos_gui.mm)
+    target_link_libraries(mujoco_simulate PUBLIC "-framework Cocoa")
+  endif()
   target_include_directories(mujoco_simulate PUBLIC ${MUJOCO_SIMULATE_DIR} ${MUJOCO_INCLUDE_DIR})
   target_link_libraries(mujoco_simulate PRIVATE lodepng mujoco::platform_ui_adapter ${MUJOCO_LIB})
   target_compile_options(mujoco_simulate PRIVATE ${MUJOCO_SILENCE_COMPILER_WARNINGS})
